### ISSUE TITLE
Add Linux to CI matrix

### ISSUE THREAD

**Issue by sudara** [sudara/melatonin_inspector#93]
See #92

**Comment by tobiashienzsch** [comment]
2 More minor issues:

## Shadow

The ctor argument `edgeGap` for `ColourPropertyComponent::ColorSelector` shadows a member in `juce::ColourSelector`. 

```sh
/home/user/melatonin_inspector/melatonin/components/colour_property_component.h: In constructor 'melatonin::ColourPropertyComponent::ColorSelector::ColorSelector(int, int, int)':
/home/user/melatonin_inspector/melatonin/components/colour_property_component.h:49:21: error: declaration of 'edgeGap' shadows a member of 'melatonin::ColourPropertyComponent::ColorSelector' [-Werror=shadow]
   49 |                 int edgeGap = 4,
      |                 ~~~~^~~~~~~~~~~
/home/user/melatonin_inspector/cmake-build-gcc/_deps/juce-src/modules/juce_gui_extra/misc/juce_ColourSelector.h:154:9: note: shadowed declaration is here
  154 |     int edgeGap;
      |         
```

## Redudant Decl

Not sure why this is coming up, as the file is auto-generated.

```sh

/home/user/melatonin_inspector/LatestCompiledAssets/BinaryData1.cpp:156:13: error: redundant redeclaration of 'const char* InspectorBinaryData::getNamedResourceOriginalFilename(const char*)' in same scope [-Werror=redundant-decls]
  156 | const char* getNamedResourceOriginalFilename (const char* resourceNameUTF8);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/melatonin_inspector/LatestCompiledAssets/InspectorBinaryData.h:80:17: note: previous declaration of 'const char* InspectorBinaryData::getNamedResourceOriginalFilename(const char*)'
   80 |     const char* getNamedResourceOriginalFilename (const char* resourceNameUTF8);
      | 

```

**Comment by baconpaul** [comment]
I temporarily addressed this in #100 by not doing warning-as-error on the linux test build and then fixing the rest of the issues

I suggest you keep this issue open and change title to "turn on warning error linux ci"

**Comment by baconpaul** [comment]
Ahh ignore that comment; with the push above I have CI running with warnings are errors on linux too.
